ChaosMod/Effects/db/Player/PlayerBlimpStrats.cpp

```diff
@@ -1,24 +1,68 @@
 /*
-	Effect by Last0xygen
+	Effect by Last0xygen, modified by Reguas
 */
 
 #include <stdafx.h>
 #include "Memory/Hooks/ScriptThreadRunHook.h"
 
 static void OnStart()
 {
+	bool cutscenePlaying = IS_CUTSCENE_PLAYING();
+
 	Hash blimpHash = GET_HASH_KEY("blimp");
+	Hash daveHash = GET_HASH_KEY("ig_davenorton");
+
 	LoadModel(blimpHash);
+	
+	if (!cutscenePlaying)
+	{
+		REQUEST_CUTSCENE("fbi_1_int", 8);
+	}
+	
 	Hooks::EnableScriptThreadBlock();
-	Vehicle veh = CREATE_VEHICLE(blimpHash, -377.276, 1055.06, 340.962, 80, true, false, false);
+	Vehicle veh = CREATE_VEHICLE(blimpHash, -370.490, 1029.085, 345.090, 53.824, true, false, false);
 	SET_VEHICLE_ENGINE_ON(veh, true, true, false);
 	Ped player = PLAYER_PED_ID();
 	SET_ENTITY_INVINCIBLE(player, true);
 	SET_PED_INTO_VEHICLE(player, veh, -1);
-	SET_VEHICLE_FORWARD_SPEED(veh, 36);
+	SET_VEHICLE_FORWARD_SPEED(veh, 45);
 	TASK_LEAVE_VEHICLE(player, veh, 4160);
-	WAIT(3000);
+
+	
+
+	int waited = 0;
+
+	while (!IS_PED_GETTING_UP(player) && waited < 100)
+	{
+		WAIT(100);
+		waited++;
+	}
+
 	SET_ENTITY_INVINCIBLE(player, false);
+
+	if (!cutscenePlaying)
+	{
+		REQUEST_ANIM_DICT("missfbi1leadinout");
+		REQUEST_MODEL(daveHash);
+
+		while (!HAS_CUTSCENE_LOADED()) // for proper cutscene play
+		{
+			WAIT(0);
+		}
+
+		REGISTER_ENTITY_FOR_CUTSCENE(player, "MICHAEL", 0, 0, 64);
+		
+		START_CUTSCENE(0);
+		WAIT(8500);
+		STOP_CUTSCENE_IMMEDIATELY();
+
+		REMOVE_CUTSCENE();
+
+		Ped pedDave = CreatePoolPed(4, daveHash, -442.2f, 1059.25f, 326.66f, 180.6f);
```

Comment:
Would be better to just spawn a normal ped and use `SET_PED_AS_NO_LONGER_NEEDED` instead, no need to make Dave persistent

Reply:
Gotcha